### ISSUE TITLE
some part still ignored

### ISSUE THREAD

**Issue by Dj-jom2x** [javascript-obfuscator/javascript-obfuscator#391]
```
       }, []), t["\u0275mpd"](1024, c.m, function() {
                    return [
                        [{
                            path: "a",
                            component: C,
                            canActivate: [bl.c],
                            data: yl
                        }, {
                            path: "b",
                            component: O,
                            canActivate: [bl.c],
                            data: Cl
                        }, {
                            path: "c",
                            component: q,
                            canActivate: [bl.c],
                            data: xl
                        }, {
                            path: "d",
                            component: B,
                            canActivate: [bl.c],
                            data: _l
                        }]
                    ]
                }, [])])
```

what options should i used to hide those... btw im using angular 8.. i tried stringArray not working  and transformObjectKeys,  rotateStringArray

**Comment by zamotkin** [comment]
If you still have a problem, could you rephrase your question?

**Comment by Dj-jom2x** [comment]
I want those path,component,canActivate,data unreadable

**Comment by zamotkin** [comment]
I see your point, will review possible implementations.

**Comment by sanex3339** [comment]
`transformObjectKeys` does not work currently with a returned array. I will see into this.

**Comment by sanex3339** [comment]
Will create a PR with improved `transformObjectKeys` later today

**Comment by Dj-jom2x** [comment]
thank you.. :) very useful for angular

**Comment by sanex3339** [comment]
Fix already in `master` branch. Can you install obfuscator directly from `master` branch and check for any runtime errors in your obfuscated code?

**Comment by Dj-jom2x** [comment]
how can I install this directly from master?

**Comment by sanex3339** [comment]
Replace version in `package.json` on `git+https://git@github.com/javascript-obfuscator/javascript-obfuscator.git`

**Comment by Dj-jom2x** [comment]
wait im using gulp file looks like this

```

gulp.task('obfuscate', function () {
  return gulp
    .src(['dist/aaa/*.js', '!dist/aaa/vendor.*.js', '!dist/aaa/polyfills.*.js'])
    .pipe(
      javascriptObfuscator({
        compact: true,
        config: '',
        controlFlowFlattening: false,
        controlFlowFlatteningThreshold: 0.75,
        deadCodeInjection: false,
        deadCodeInjectionThreshold: 0.4,
        debugProtection: true,
        debugProtectionInterval: true,
        disableConsoleOutput: false,
        domainLock: [],
        exclude: [
          'common.js',
          'main.js',
          'scripts.js',
          'scripts.js',
          'vendor.js'
        ],
        identifiersPrefix: 'kishaki_',
        inputFileName: '',
        log: false,
        renameGlobals: false,
        reservedNames: [],
        reservedStrings: [],
        rotateStringArray: false,
        seed: 0,
        selfDefending: true,
        sourceMap: false,
        sourceMapBaseUrl: '',
        sourceMapFileName: '',
        stringArray: false,
        stringArrayEncoding: false,
        stringArrayThreshold: 0.75,
        transformObjectKeys: false,
        unicodeEscapeSequence: false
      })
    )
    .pipe(gulp.dest('dist-obfs'));
})


gulp.task('default', gulp.parallel('obfuscate'));

```

gulp-javascript-obfuscator

i did npm install https://github.com/javascript-obfuscator/javascript-obfuscator.git
then npm install gulp-javascript-obfuscator

is gulp-javascript-obfuscator wrapper? else i cant see the new patch

**Comment by sanex3339** [comment]
With gulp, better to wait for the release later today.

**Comment by Dj-jom2x** [comment]
alright. I wait because using this is neat.

**Comment by sanex3339** [comment]
New version `0.21.0` is out

**Comment by Dj-jom2x** [comment]
i will try it now

**Comment by Dj-jom2x** [comment]
its working pretty good.. but let me mess on the settings because i get this..

![image](https://user-images.githubusercontent.com/12135498/71851192-f2257400-3110-11ea-976b-a3d3c05d4e62.png)

**Comment by Dj-jom2x** [comment]
whats the best config for angular XD

**Comment by Dj-jom2x** [comment]
even just debugProtection still detected.

**Comment by sanex3339** [comment]
What part is still need hiding?

**Comment by Dj-jom2x** [comment]
like how the router works xD and guards. with tons of lazy route configs.

**Comment by Dj-jom2x** [comment]
what settings did you use on your angular project? https://www.virustotal.com/gui/file-analysis/NzE0ZjU0Nzc0ZmExMmEwYzJmZGU0OGU2MTk4NjZkYjA6MTU3ODUzMDU0Mw==/detection

https://gist.github.com/sanex3339/ffc2876123b52e6d11ce45369fd53acf

**Comment by sanex3339** [comment]
This project used angular@2

Try this
https://github.com/javascript-obfuscator/javascript-obfuscator#antiviruses-false-positive-virus-alerts
